__tests__/app.test.js

```diff
@@ -192,10 +192,95 @@ describe('PATCH /api/articles', () => {
         const updatedVote = {}
         return request(app).patch(`/api/articles/4`)
-        .send(updatedVote)
+            .send(updatedVote)
+            .expect(400)
+            .then(({
+                body
+            }) => {
+                expect(body).toEqual({
+                    msg: "bad request"
+                })
+            })
+    });
+})
+
+describe('GET /api/articles', () => {
+    it('returns an array of article objects with the correct properties', () => {
+        return request(app)
+            .get(`/api/articles`)
+            .expect(200)
+            .then(({
+                body
+            }) => {
+                const {
+                    articles
+                } = body
+                expect(articles).toBeInstanceOf(Array)
+                expect(articles.forEach((article) => {
+                    expect(article).toEqual(expect.objectContaining({
+                        article_id: expect.any(Number),
+                        title: expect.any(String),
+                        topic: expect.any(String),
+                        author: expect.any(String),
+                        body: expect.any(String),
+                        created_at: expect.any(String),
+                        votes: expect.any(Number),
+                        comment_count: expect.any(Number),
+                    }))
+                }))
+            })
+    })
+    it('returns the articles by date in descending order when no sort_by or sortOrder is provided', () => {
+        return request(app).get('/api/articles')
+        .expect(200)
+        .then(({body}) => {
+            const { articles } = body
+            expect(articles).toBeSortedBy('created_at', {descending: true})
+        })
+    });
+    it('should return a 400 error when sort_By is not an existing validRow', () => {
+        return request(app)
+        .get('/api/articles?sort_by=notcollumn')
+        .expect(400)
+        .then(({body}) => {
+            expect(body.msg).toBe('bad request')
+        })
+    })
+    it('should return a 400 error when sort_Order is not an existing validOrder', () => {
+        return request(app)
+        .get('/api/articles?order=gdfgf')
         .expect(400)
         .then(({body}) => {
-            expect(body).toEqual({msg : "bad request"})
+            expect(body.msg).toBe('bad request')
+        })
+    })
+    it('returns all objects sorted by the provided column name', () => {
+        return request(app)
+        .get('/api/articles?sort_by=article_id')
+        .expect(200)
+        .then(({ body })=>{
+            const { articles } = body
+            expect(articles).toBeSortedBy('article_id', {descending: true})
+        })
+    })
+    it('returns array of articles sorted by descending order with specified topic', () => {
+        const query = '?topic=cats'
+        return request(app).get(`/api/articles${query}`)
+        .expect(200)
```

Comment:
Id do a length check to make sure you get back all the ones your expect for this topic